frontend/lib/src/theme/utils.ts

```diff
@@ -182,6 +182,69 @@ export const parseRadius = (
   return [radiusValue, cssUnit]
 }
 
+/**
+ * Helper function to parse fontSize options which allow the same possible values
+ * @param fontSize a string number in pixels or rem; handles number values as pixels
+ * (e.g. "15px", "0.875rem", "15")
+ * @returns font size in em (e.g. "0.875em")
+ */
+export const parseFontSize = (
+  configName: string,
+  fontSize: string | number,
+  inSidebar: boolean
+): string | undefined => {
+  const themeSection = inSidebar ? "theme.sidebar" : "theme"
+
+  if (typeof fontSize === "string") {
+    // If string, check its valid (ends with "rem" or "px")
+    // and can be parsed as a number
+    const processedFontSize = fontSize.trim().toLowerCase()
+    const parsedFontSize = parseFloat(processedFontSize)
+    if (
+      parsedFontSize &&
+      (processedFontSize.endsWith("rem") || processedFontSize.endsWith("px"))
+    ) {
+      return processedFontSize
+    }
+
+    // Fallback: if the value can be parsed as a number, treat it as pixels
+    if (parsedFontSize.toString() === processedFontSize) {
+      return `${processedFontSize}px`
+    }
+  }
+  // If invalid, log warning and return undefined
+  LOG.warn(
+    `Invalid size passed for ${configName} in ${themeSection}: ${fontSize}. Falling back to default ${configName}.`
+  )
+}
+
+/**
+ * Calculate an em value for inline code based on the configured code font size
+ * This ensures inline code scales properly relative to its parent while respecting the configured size
+ * @param codeFontSize: the configured code font size (e.g. "14px", "0.875rem")
+ * @param baseFontSize: the base font size in pixels
+ * @returns em value as a string (e.g. "0.875em")
+ */
+export const calculateInlineCodeFontSize = (
+  codeFontSize: string,
+  baseFontSize: number
+): string => {
+  const processedFontSize = codeFontSize.trim().toLowerCase()
+
+  if (processedFontSize.endsWith("rem")) {
+    // Simply replace "rem" with "em" since both are relative units
+    return processedFontSize.replace("rem", "em")
+  } else if (processedFontSize.endsWith("px")) {
+    // Convert px to em using baseFontSize
+    const emValue = parseFloat(processedFontSize) / baseFontSize
+    return `${emValue.toFixed(3)}em`
+  }
+
+  // This should never happen (parseFontSize has already returned px/rem value for
+  // parsedCodeFontSize, this called after checking parsedCodeFontSize not undefined)
+  return "0.75em"
```

Comment:
nit: slightly better here might be to directly return the primitives value from typography, e.g.:

```typescript
import { fontSizes } from "./primitives/typography"

...

fontSizes.inlineCodeFontSize
```